### PR TITLE
search bar in glossary

### PR DESCRIPTION
**Pull request by fadeomar**
#181

### CODE REVIEW

File: client/src/components/pages/Glossary/index.js [hunk 1]
```diff
@@ -1,26 +1,59 @@
...
-};
+const { Search } = Input;
+
+class Glossary extends Component {
+  state = {
+    searchValue: '',
+    items: [],
+  };
+
+  onChange = v => {
```
Comment (AbdallahAmmar96) [inline client/src/components/pages/Glossary/index.js]: use destruction

File: client/src/components/pages/Glossary/style.css [hunk 2]
```diff
@@ -1,6 +1,7 @@
 .glossary_container {
   background: #E0E0E0;
   width: 100vw;
+  min-height:100vh ;
```
[the remark this answers is not in the document]
Comment (fadeomar) [reply client/src/components/pages/Glossary/style.css]: Done

File: client/src/components/pages/Glossary/index.js [hunk 3]
```diff
@@ -1,26 +1,59 @@
...
+    const searchValue = v.target.value.toUpperCase();
+    const items = glossaryData.filter(item =>
+      Object.keys(item)[0].includes(searchValue)
+    );
+    this.setState(() => ({ items, searchValue }));
```
[the remark this answers is not in the document]
Comment (fadeomar) [reply client/src/components/pages/Glossary/index.js]: done

File: client/src/components/pages/Glossary/style.css [hunk 4]
```diff
@@ -10,6 +11,11 @@
   margin-bottom: 1rem;
 }
 
+.ant-input-search {
+  width: 95% !important;
+  margin-bottom: 20px;
```
Comment (MohammadAlhalaq) [inline client/src/components/pages/Glossary/style.css]: You need to use rem or px :)

File: client/src/components/pages/Glossary/style.css [hunk 5]
```diff
@@ -1,6 +1,7 @@
 .glossary_container {
   background: #E0E0E0;
   width: 100vw;
+  min-height:100vh;
```
Comment (MohammadAlhalaq) [inline client/src/components/pages/Glossary/style.css]:
```suggestion
  min-height: 100vh;
```

Comment (fadeomar) [reply client/src/components/pages/Glossary/style.css]: Done

File: client/src/components/pages/Glossary/style.css (hunk 4 above)
Comment (fadeomar) [reply client/src/components/pages/Glossary/style.css]: Done

File: client/src/components/pages/Glossary/index.js (hunk 1 above)
Comment (fadeomar) [reply client/src/components/pages/Glossary/index.js]: Done

File: client/src/components/pages/Glossary/index.js [hunk 6]
```diff
@@ -1,26 +1,59 @@
...
+const { Search } = Input;
+
+class Glossary extends Component {
+  state = {
+    searchValue: '',
+    items: [],
+  };
+
+  onChange = v => {
+    const searchValue = v.target.value.toUpperCase();
```
[the remark this answers is not in the document]
Comment (fadeomar) [reply client/src/components/pages/Glossary/index.js]: Done

File: client/src/components/pages/Glossary/index.js [hunk 7]
```diff
@@ -1,26 +1,59 @@
...
+    const items = glossaryData.filter(item =>
+      Object.keys(item)[0].includes(searchValue)
```
[the remark this answers is not in the document]
Comment (fadeomar) [reply client/src/components/pages/Glossary/index.js]: Done

File: client/src/components/pages/Glossary/index.js [hunk 8]
```diff
@@ -1,26 +1,81 @@
...
+  }
+
+  filterItems = () => {
+    const { searchValue } = this.state;
+
+    let items = glossaryData.filter(item =>
+      Object.keys(item)[0]
+        .toLowerCase()
+        .includes(searchValue)
+    );
```
Comment (MohammadAlhalaq) [inline client/src/components/pages/Glossary/index.js]:
```javascript
componentDidUpdate(prevProps, prevState) {
    const { searchValue } = this.state;
    if (prevState.searchValue !== searchValue) this.filterItems(searchValue);
  }

  filterItems = (keyForFilter) => {

    let items = glossaryData.filter(item =>
      Object.keys(item)[0]
        .toLowerCase()
        .includes(keyForFilter)
    );
```

**Pull request merged**